Scala3 Documentation (#3)
scalacenter/docs.scala-lang: _overviews/scala3-book/taste-intro.md

---
title: A Taste of Scala
type: chapter
description: This chapter provides a high-level overview of the main features of the Scala 3 programming language.
num: 4
previous-page: why-scala-3
next-page: taste-hello-world
---

This “Taste of Scala” chapter provides a whirlwind tour of the main features of the Scala 3 programming language. After the initial tour in this chapter, the rest of the book provides more details on these features, and the [Reference documentation][reference] provides _many_ more details.

> >Throughout this book, you’ll be able to test many of the examples directly on this page. In addition to that, you can also test anything you’d like on [ScalaFiddle.io](https://scalafiddle.io), [Scastie](https://scastie.scala-lang.org), or in the Scala REPL, which is demonstrated shortly.

Author: It is actually not true that one can test examples on the page (yet?).
```suggestion
>Throughout this book, we recommend to experiment with the examples on [Scastie](https://scastie.scala-lang.org), or in the Scala REPL, which is demonstrated shortly.
```
